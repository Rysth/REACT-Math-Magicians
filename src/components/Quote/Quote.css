.quote {
  text-align: center;
}

.quote-description {
  font-size: 0.9rem;
  font-style: italic;
}

.quote-description.author {
  font-weight: bold;
  margin: 0;
}

.quote-loading {
  font-size: 1.5rem;
  font-weight: bold;
  color: var(--CL-primary-orange);
  filter: drop-shadow(0 0 1px rgb(0, 0, 0, 0.5));
}
