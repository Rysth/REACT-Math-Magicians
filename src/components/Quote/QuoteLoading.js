import PropTypes from 'prop-types';
import { useEffect, useState } from 'react';

export default function QuoteLoading({ timer }) {
  const [message, setMessage] = useState('Loading');

  useEffect(() => {
    const interval = setInterval(() => {
      setMessage((prevMessage) => {
        if (prevMessage === 'Loading...') {
          clearInterval(interval);
          return prevMessage;
        }
        return prevMessage.concat('.');
      });
    }, timer);

    return () => {
      clearInterval(interval);
    };
  }, [timer]);

  return <p className="quote-loading">{message}</p>;
}

QuoteLoading.defaultProps = {
  timer: 1000,
};

QuoteLoading.propTypes = {
  timer: PropTypes.number,
};
