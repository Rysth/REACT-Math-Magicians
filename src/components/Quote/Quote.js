import { useEffect, useState } from 'react';
import QuoteLoading from './QuoteLoading';
import './Quote.css';

const API = 'https://api.api-ninjas.com/v1/quotes';

export default function Quote() {
  const [quote, setQuote] = useState();
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(false);
  useEffect(() => {
    const fetchNinjaAPI = async () => {
      try {
        const response = await fetch(API, {
          method: 'GET',
          headers: {
            'X-Api-Key': process.env.REACT_APP_API_KEY,
          },
        });
        const data = await response.json();
        setQuote({
          message: data[0]?.quote,
          author: data[0]?.author,
        });
      } catch (error) {
        setError(true);
        throw new Error('Error fetching data.');
      }
    };

    fetchNinjaAPI();

    setTimeout(() => {
      setLoading(true);
    }, 1500);
  }, []);

  return (
    <div className="quote">
      {!loading && !error ? (
        <QuoteLoading timer={500} />
      ) : (
        <>
          <p className="quote-description">{quote ? quote.message : ''}</p>
          <p className="quote-description author">{quote ? quote.author : ''}</p>
        </>
      )}
    </div>
  );
}
