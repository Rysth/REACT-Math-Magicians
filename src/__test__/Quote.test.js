import renderer from 'react-test-renderer';
import Quote from '../components/Quote/Quote';

describe('Qoute Component',()=>{
    it('Should create a Qoute component Snapshot',()=>{
        const tree = renderer.create(<Quote/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})