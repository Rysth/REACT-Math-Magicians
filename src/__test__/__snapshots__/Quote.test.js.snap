// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Qoute Component Should create a Qoute component Snapshot 1`] = `
<div
  className="quote"
>
  <p
    className="quote-loading"
  >
    Loading.
  </p>
</div>
`;
